rust-toolchain.toml

```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel = "nightly-2024-04-03"
+components = ["rustfmt", "clippy"]
```

Comment:
If we're touching this anyways, should we also add `rust-analyzer`? It shouldn't cause any problems, just increased disk usage for people who might not want it. It's a bit annoying for me on every toolchain update to have to manually enable this component.

Reply:
Are you on vim/emacs? I use zed/vscode which I _believe_ both just bundle it themselves.